script.js

```diff
@@ -1 +1,402 @@
-console.log('yay!');
+// Global setup ----------------------------------------------------
+let board = [];
+let firstCard = null;
+const boardSize = 4; // has to be an even number
+const matchesNeededToWin = (boardSize * boardSize) / 2;
+let numOfMatches = 0;
+let deck;
+let playerName;
+let secondsLeftTillGameOver = 180;
+let numOfGames = 1;
+const loseMessage = 'Sorry but the time ran out! Click on a square to try again!';
+let timerId; // to store ID of startMatchTimer
```

Comment:
Globals can be grouped together by type, with more comments to explain what they will be used for.